### PR TITLE
Destroy PlayerInfo after EngineRecord is stopped to fix a debug assertion

### LINKED ISSUE #12242: DEBUG ASSERT on shutdown: PlayerInfo::instance()
### Bug Description

```
DEBUG ASSERT: "s_pPlayerInfo" in function static PlayerInfo& PlayerInfo::instance() at /home/uk/volumes/Build/cpp/mixxx/src/mixer/playerinfo.cpp:41
```

```
#8  0x00000000006b5985 in mixxx_debug_assert(char const*, char const*, int, char const*) (assertion=<optimized out>, file=<optimized out>, line=<optimized out>, function=<optimized out>)
    at /home/uk/volumes/Build/cpp/mixxx/src/util/assert.h:9
#9  0x0000000000bfaa25 in mixxx_debug_assert_return_true(char const*, char const*, int, char const*)
    (function=0x602bd0 "static PlayerInfo& PlayerInfo::instance()", line=41, file=0x66fa90 "/home/uk/volumes/Build/cpp/mixxx/src/mixer/playerinfo.cpp", assertion=0x649b2c "s_pPlayerInfo")
    at /home/uk/volumes/Build/cpp/mixxx/src/util/assert.h:18
#10 PlayerInfo::instance() () at /home/uk/volumes/Build/cpp/mixxx/src/mixer/playerinfo.cpp:41
#11 0x0000000000967d44 in EngineRecord::metaDataHasChanged() (this=this@entry=0x213d8b0) at /home/uk/volumes/Build/cpp/mixxx/src/engine/sidechain/enginerecord.cpp:78
#12 0x000000000096b6a8 in EngineRecord::process(float const*, int) (this=0x213d8b0, pBuffer=0x7f33fff00040, iBufferSize=36864)
    at /home/uk/volumes/Build/cpp/mixxx/src/engine/sidechain/enginerecord.cpp:207
#13 0x000000000096e3d2 in EngineSideChain::run() (this=0x1de12f0) at /home/uk/volumes/Build/cpp/mixxx/src/engine/sidechain/enginesidechain.cpp:117
```

### Version

main + 2.4 / Qt5

### OS

Fedora 38

### PR DESCRIPTION
**Pull request by daschuer**
This fixes #12242

### CODE REVIEW

File: src/coreservices.cpp [hunk 1]
```diff
@@ -594,6 +589,12 @@ void CoreServices::finalize() {
     qDebug() << t.elapsed(false).debugMillisWithUnit() << "deleting EngineMixer";
     CLEAR_AND_CHECK_DELETED(m_pEngine);
 
+    // Destroy PlayerInfo explicitly to release the track
+    // pointers of tracks that were still loaded in decks
+    // or samplers when PlayerManager was destroyed!
+    // After EngineMixxer which makes use of PlayerInfo in EngineRecord #12242
```
Comment (ronso0) [inline src/coreservices.cpp:595]:
```suggestion
    // Do this after deleting EngineMixer which makes use of
    // PlayerInfo in EngineRecord. Fixes #12242
```

**Review by ronso0 — COMMENTED** [review]
LGTM, just the comment can be improved a bit.

**Comment by ronso0** [comment]
@uklotzde please take a look.

File: src/coreservices.cpp [hunk 2]
```diff
@@ -594,6 +589,13 @@ void CoreServices::finalize() {
     qDebug() << t.elapsed(false).debugMillisWithUnit() << "deleting EngineMixer";
     CLEAR_AND_CHECK_DELETED(m_pEngine);
 
+    // Destroy PlayerInfo explicitly to release the track
+    // pointers of tracks that were still loaded in decks
+    // or samplers when PlayerManager was destroyed!
+    // Do this after deleting EngineMixer which makes use of
+    // PlayerInfo in EngineRecord. Fixes #12242
```
Comment (uklotzde) [inline src/coreservices.cpp:596]: References to closed issues should not be needed to reason about code. Comments and code should be self-sufficient, which is the case here. That suffix doesn't provide any value. If you want to keep it in the history then add in to the Git commit message.

TODO comments with references to open issues, discussions, or whatever are desirable, though. But then better use the full URL.

**Comment by daschuer** [comment]
Done

**Comment by ronso0** [comment]
Thanks. Squash, then merge?

**Comment by daschuer** [comment]
Squashed and ready for merge

**Pull request merged**